src/main/java/com/pingcap/tikv/policy/RetryNTimes.java

```diff
@@ -18,34 +18,44 @@
 import com.google.common.base.Preconditions;
 import com.pingcap.tikv.operation.ErrorHandler;
 
-public class RetryNTimes extends RetryPolicy {
-  private int n;
+public class RetryNTimes<T> extends RetryPolicy<T> {
+  private int[] fibonacci;
 
-  private RetryNTimes(int n, ErrorHandler handler) {
+  private RetryNTimes(int n, ErrorHandler<T> handler) {
     super(handler);
     Preconditions.checkArgument(n >= 1, "Retry count cannot be less than 1.");
-    this.n = n;
+    this.fibonacci = generateFibonacci(n);
   }
 
   @Override
-  protected boolean shouldRetry(Exception e) {
-    return --n != 0;
+  public int[] getFibonacci() {
```

Comment:
If backoff strategy is implemented in parent class, please put this in parent.
If you decide to make it in subclass, you might move callWithRetry here.
A single logic is better be contained with in a single class for clear design.

Reply:
I am rewriting it. Original design is not so good.